src/lib/gadgets/elliptic-curve.ts

```diff
@@ -258,6 +319,11 @@ function multiScalarMul(
     for (let i = 0; i < n; i++) {
       sum = Curve.add(sum, Curve.scale(P[i], s[i]));
     }
+    if (mode === 'assert-zero') {
+      assert(sum.infinity, 'scalar multiplication: expected zero result');
+      return Point.from(Curve.zero);
+    }
+    assert(!sum.infinity, 'scalar multiplication: expected non-zero result');
```

Comment:
added a different mode for the scalar mul gadget so that we can also support the subgroup check (q*P = 0) with it